jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java

```diff
@@ -457,6 +460,30 @@ public void matchCreated(MatchCreatedEvent event) {
                             || ruleName.startsWith( "RuleFlow-AdHocActivate-" )) {
                         kruntime.queueWorkingMemoryAction(new SignalManagerSignalAction(ruleName, event));
                     }
+                } else {
+                    String ruleName = event.getMatch().getRule().getName();
+                    if ( ruleName.startsWith( "RuleFlow-Start-" )) {
+                        String processId = ruleName.replace("RuleFlow-Start-", "");
+                        RuntimeManager manager = (RuntimeManager) kruntime.getEnvironment().get(EnvironmentName.RUNTIME_MANAGER);
+                        if (manager != null) {
+                            org.kie.api.runtime.manager.Context<?> context = ProcessInstanceIdContext.get();
+                            
+                            String caseId = (String) kruntime.getEnvironment().get(EnvironmentName.CASE_ID);
+                            if (caseId != null) {
+                                context = CaseContext.get(caseId);
+                            }
+                            
+                            RuntimeEngine runtime = manager.getRuntimeEngine(context);
+                            KieSession ksession = runtime.getKieSession();
+                            try {
+                                ksession.execute(new StartProcessWithTypeCommand(processId, null, "conditional"));
+                            } finally {
+                                manager.disposeRuntimeEngine(runtime);
+                            }
+                        } else {
+                            startProcess( processId, null, "conditional");
+                        }
```

Comment:
Looks good but I have just noticed that this part of code (starting with `if (manager != null)`) is exactly the same as this part here:
https://github.com/kiegroup/jbpm/blob/4afa414e0159e5727bccb358323905e363ef7136/jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java#L417-L435
So maybe it might be good to export it to method.

Reply:
good point, extracted to new method

Reply:
BTW why does it have to be done separately (in case there is / is not a runtime manager available) when in the end it calls the same method? We just want to be sure that we call it via right runtime manager in case of PPI strategy for example? Thanks.

Reply:
th reason is to use the right context - like to not use the same ksession for new process instance but start a new one - mainly for strategies different than singleton

Reply:
Hmm. So what if it is a Singleton strategy - do we want to create PPIContext anyway?

Reply:
that does not matter as singleton does not care about the context, it has always loaded single instance of runtime engine